exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer_test.go

```diff
@@ -282,3 +283,177 @@ func TestEnrichMetricNamesWithAttributes(t *testing.T) {
 		})
 	}
 }
+
+func TestComputeRateMetricFromCounter(t *testing.T) {
+	t.Parallel()
+	producer := &MetricsProducer{
+		logger:           nil,
+		previousCounters: make(map[string]BMCHelixOMSample),
+	}
+
+	labels := map[string]string{
+		"entityId":   "OTEL:host:network:eth0",
+		"metricName": "hw.network.io",
+		"unit":       "By",
+		"hostname":   "host",
+		"source":     "OTEL",
+	}
+
+	// First sample: initial counter datapoint
+	first := BMCHelixOMMetric{
+		Labels: labels,
+		Samples: []BMCHelixOMSample{{
+			Value:     5000,
+			Timestamp: time.Now().UnixMilli(),
+		}},
+	}
+
+	// First call – no rate should be returned yet
+	assert.Nil(t, producer.computeRateMetricFromCounter(first), "First datapoint should not yield a rate metric")
+
+	// Simulate a second datapoint after a short delay (simulate time passage)
+	time.Sleep(20 * time.Millisecond)
```

Comment:
Do we need to wait for real time to pass, or could this be as simple as:

```
now := time.Now()
// ...
next := now.Add(100* time.Milliseconds)
```

Reply:
Yes, your suggestion is better 😁